### ISSUE TITLE
pmd script fails

### ISSUE THREAD

**Issue by RuslanGaliullin** [yegor256/cam#151]
get log
```
/usr/local/bin/pmd: line 231: 242911 Killed                  java ${HEAPSIZE} ${PMD_JAVA_OPTS} $(jre_specific_vm_options) -cp "${classpath}" "${CLASSNAME}" "$@"
Failed to collect pmd.sh for /dataset/github/lenve/JavaEETest/Test8-EL/src/main/java/org/sang/Main.java
```

for calculating pmd metric for Main.java:
```
package org.sang;

import org.springframework.context.annotation.AnnotationConfigApplicationContext;

/**
 * Created by sang on 16-12-13.
 */
public class Main {
    public static void main(String[] args) {
        AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(MyConfig.class);
        ELConfig bean = context.getBean(ELConfig.class);
        bean.output();
        context.close();
    }
}
```

**Comment by RuslanGaliullin** [comment]
@yegor256

**Comment by yegor256** [comment]
@RuslanGaliullin is `pmd` installed?

**Comment by RuslanGaliullin** [comment]
> @RuslanGaliullin is `pmd` installed?

yeah, it even works for other files around

**Comment by RuslanGaliullin** [comment]
Actually adding `make install` solved #145 issue for most of the files, however this problem with pmd still exists.

One more example

```
/usr/local/bin/pmd: line 231: 3297674 Killed                  java ${HEAPSIZE} ${PMD_JAVA_OPTS} $(jre_specific_vm_options) -cp "${classpath}" "${CLASSNAME}" "$@"
Failed to collect pmd.sh for /dataset/github/apache/camel/components/camel-resilience4j/src/main/java/org/apache/camel/component/resilience4j/ResilienceReifier_configureBulkHead_3.java
4 scripts collected 9 metrics (sum=7455.6) for ResilienceReifier_configureBulkHead_3.java (101226/613218), in 38s
```
ResilienceReifier_configureBulkHead_3.java:
```
class ResilienceReifier_configureBulkHead_3 {public BulkheadConfig configureBulkHead(Resilience4jConfigurationCommon config) {
        if (!parseBoolean(config.getBulkheadEnabled(), false)) {
            return null;
        }

        BulkheadConfig.Builder builder = BulkheadConfig.custom();
        if (config.getBulkheadMaxConcurrentCalls() != null) {
            builder.maxConcurrentCalls(parseInt(config.getBulkheadMaxConcurrentCalls()));
        }
        if (config.getBulkheadMaxWaitDuration() != null) {
            long duration = parseLong(config.getBulkheadMaxWaitDuration());
            if (duration <= 0) {
                builder.maxWaitDuration(Duration.ZERO);
            } else {
                builder.maxWaitDuration(Duration.ofMillis(duration));
            }
        }
        return builder.build();
    }}
```
@yegor256

**Comment by yegor256** [comment]
@RuslanGaliullin this looks like "out of memory" problem -- your OS kills Java that is taking too much resources (since we run many PMD instances at the same time). Maybe we can introduce a limitation on the amount of parallel processes to run: https://github.com/yegor256/cam/blob/master/help/parallel.sh#L30  Here, `--max-procs` is using the result of `nproc`, which may be pretty high (especially if multiplied).
